File: 2_mpi/mpi-intro/hello-world.f90
program helloworld
use mpi
implicit none
integer :: rank, comsize, ierr

call MPI_Init(ierr)
call MPI_Comm_size(MPI_COMM_WORLD, comsize, ierr)
call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)
print *,"Hello, world from task ",rank, &
        ' of ', comsize

call MPI_Finalize(ierr)
end program helloworld

File: code/2_mpi/mpi-intro/hello-world.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv) {
	
	int rank, size;
	int ierr;

	ierr = MPI_Init(&argc, &argv);
	
	ierr = MPI_Comm_size(MPI_COMM_WORLD, &size);
	ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank);

	printf("Hello, world from task %d of %d!\n",rank,size); 

	MPI_Finalize();

	return ierr;
}
